erpnext/hr/doctype/salary_component/salary_component.js

```diff
@@ -2,7 +2,17 @@
 // For license information, please see license.txt
 
 frappe.ui.form.on('Salary Component', {
-	refresh: function(frm) {
-
+	setup: function(frm) {
+		frm.set_query("default_account", "accounts", function(doc, cdt, cdn) {
+			var d = locals[cdt][cdn];
+			var account_types = ["Bank", "Cash"];
```

Comment:
Account type linked to salary company is not bank / cash, it should be Expense account (root_type="Expense")